Extensions/BBText/JsExtension.js

```diff
@@ -0,0 +1,511 @@
+        objectContent.visible = newValue === '1';
+        return true;
+      }
+      if (propertyName === 'size') {
+        objectContent.size = String(newValue);
+        return true;
+      }
+      if (propertyName === 'align') {
+        objectContent.align = newValue ? newValue : 'left';
+        return true;
+      }
+
+      return false;
+    };
+    objectBBText.getProperties = function(objectContent) {
+      var objectProperties = new gd.MapStringPropertyDescriptor();
+
+      objectProperties.set(
+        'text',
+        new gd.PropertyDescriptor(objectContent.text)
+          .setType('textarea')
+          .setLabel(_('BBCode text'))
+      );
+
+      objectProperties.set(
+        'color',
+        new gd.PropertyDescriptor(objectContent.color)
+          .setType('color')
+          .setLabel(_('Base color'))
+      );
+
+      objectProperties.set(
+        'opacity',
+        new gd.PropertyDescriptor(objectContent.opacity.toString())
+          .setType('number')
+          .setLabel(_('Opacity (0-255)'))
+      );
+
+      objectProperties.set(
+        'size',
+        new gd.PropertyDescriptor(objectContent.size)
+          .setType('number')
+          .setLabel(_('Base size'))
+      );
+
+      objectProperties.set(
+        'align',
+        new gd.PropertyDescriptor(objectContent.align)
+          .setType('choice')
+          .addExtraInfo('left')
+          .addExtraInfo('center')
+          .addExtraInfo('right')
+          .setLabel(_('Base alignment'))
+      );
+
+      objectProperties.set(
+        'family',
+        new gd.PropertyDescriptor(objectContent.family)
+          .setType('string')
+          .setLabel(_('Base font family'))
+      );
+
+      objectProperties.set(
+        'visible',
+        new gd.PropertyDescriptor(objectContent.visible ? 'true' : 'false')
+          .setType('boolean')
+          .setLabel(_('Visible on start'))
+      );
+
+      return objectProperties;
+    };
+    objectBBText.setRawJSONContent(
+      JSON.stringify({
+        text:
+          '[b]bold[/b] [i]italic[/i] [size=15]smaller[/size] [font=times]times[/font] font\n[spacing=12]spaced out[/spacing]\n[outline=yellow]outlined[/outline] [shadow=red]DropShadow[/shadow] ',
+        opacity: 255,
+        size: '20',
+        visible: true,
+        color: '#000000',
+        family: 'Arial',
+        align: 'left',
+      })
+    );
+
+    objectBBText.updateInitialInstanceProperty = function(
+      objectContent,
+      instance,
+      propertyName,
+      newValue,
+      project,
+      layout
+    ) {
+      return false;
+    };
+    objectBBText.getInitialInstanceProperties = function(
+      content,
+      instance,
+      project,
+      layout
+    ) {
+      var instanceProperties = new gd.MapStringPropertyDescriptor();
+      return instanceProperties;
+    };
+
+    const object = extension
+      .addObject(
+        'BBText',
+        _('BBText'),
+        _('Displays a BBCode rich Text.'),
```

Comment:
no capital T. Also better be more explicit, it's the object description and so very important for new users:
`Displays a rich text using BBCode markup (allowing to set parts of the text as bold, italic, use different colors and shadows).`